pkg/auth/common/login_handler.go

```diff
@@ -366,21 +430,67 @@ func (h *TanzuLoginHandler) promptAndLoginWithAuthCode(ctx context.Context, auth
 	return wg.Wait
 }
 
+// Returns custom TLS configuration if explicitly provided to the handler,
+// of if persisted cert information associated with the issuer endpoint is found,
+// with the provided information taking precedence over persisted information.
+func (h *TanzuLoginHandler) getTLSConfig() *tls.Config {
+	var savedCertData string
+	var savedSkipVerify bool
+
+	c, _ := config.GetCert(h.issuer)
+
+	if c != nil {
+		savedCertData = c.CACertData
+		savedSkipVerify, _ = strconv.ParseBool(c.SkipCertVerify)
+	}
+
+	if savedSkipVerify || h.tlsSkipVerify {
+		//nolint:gosec // skipTLSVerify: true is only possible if the user has explicitly enabled it
+		return &tls.Config{InsecureSkipVerify: true, MinVersion: tls.VersionTLS12}
+	}
+
+	caCertData := savedCertData
+	if h.caCertData != "" {
+		caCertData = h.caCertData
+	}
+
+	if caCertData != "" {
+		var pool *x509.CertPool
+		var err error
+		pool, err = x509.SystemCertPool()
+		if err != nil || pool == nil {
+			pool = x509.NewCertPool()
+		}
+		pool.AppendCertsFromPEM([]byte(caCertData))
```

Comment:
We will need to base64 decode the `caCertData` and also log some warnings if decoding or AppendCertsFromPEM fails.